Utils/Game.py

```diff
@@ -1,13 +1,337 @@
+            button_color = self.GRAY
+        else:
+            button_color = self.LIGHT_GRAY
+
+        pygame.draw.rect(self.screen, button_color,
+                         (self.START_BUTTON_X, self.START_BUTTON_Y, self.BUTTON_WIDTH, self.BUTTON_HEIGHT))
+        pygame.draw.rect(self.screen, self.BLACK,
+                         (self.START_BUTTON_X, self.START_BUTTON_Y, self.BUTTON_WIDTH, self.BUTTON_HEIGHT), 2)
+        self.screen.blit(text, text_rect)
+
+    def draw_undo_button(self):
+        font = pygame.font.SysFont(None, 64)
+        text = font.render('UNDO', True, self.BLACK)
+        text_rect = text.get_rect(
+            center=(self.UNDO_BUTTON_X + self.BUTTON_WIDTH // 2, self.UNDO_BUTTON_Y + self.BUTTON_HEIGHT // 2))
+
+        mouse_pos = pygame.mouse.get_pos()
+        is_mouse_over = pygame.Rect(self.UNDO_BUTTON_X, self.UNDO_BUTTON_Y, self.BUTTON_WIDTH,
+                                    self.BUTTON_HEIGHT).collidepoint(mouse_pos)
+
+        if is_mouse_over and pygame.mouse.get_pressed()[0]:
+            button_color = self.GRAY
+        else:
+            button_color = self.LIGHT_GRAY
+
+        pygame.draw.rect(self.screen, button_color,
+                         (self.UNDO_BUTTON_X, self.UNDO_BUTTON_Y, self.BUTTON_WIDTH, self.BUTTON_HEIGHT))
+        pygame.draw.rect(self.screen, self.BLACK,
+                         (self.UNDO_BUTTON_X, self.UNDO_BUTTON_Y, self.BUTTON_WIDTH, self.BUTTON_HEIGHT), 2)
+        self.screen.blit(text, text_rect)
+
+    def draw_material_diff(self):
+        ratio = (self.gb.black_captured_value + 10) / (self.gb.black_captured_value + 20 + self.gb.white_captured_value)
+        b_x, b_y = self.X_OFFSET - 2 * self.MATERIAL_CHART_WIDTH, self.Y_OFFSET
+        w_x, w_y = b_x, b_y + int(ratio * self.BOARD_SIZE)
+        b_x_diff, b_y_diff=self.MATERIAL_CHART_WIDTH, w_y
+        w_x_diff, w_y_diff = self.MATERIAL_CHART_WIDTH, self.Y_OFFSET + self.BOARD_SIZE-w_y
+        pygame.draw.rect(self.screen, (0, 0, 0), (b_x, b_y,b_x_diff, b_y_diff ))
+
+        pygame.draw.rect(self.screen, (255, 255, 255),
+                         (b_x, w_y,w_x_diff, w_y_diff))
+
+    def draw_utils(self):
+        self.draw_reset_button()
+        self.draw_undo_button()
+        self.draw_material_diff()
+
+    def reset_clicked(self, x, y):
+        button_rect = pygame.Rect(self.START_BUTTON_X, self.START_BUTTON_Y, self.BUTTON_WIDTH,
+                                  self.BUTTON_HEIGHT)
+        if button_rect.collidepoint(x, y):
+            self.reinitialize()
+
+    # TODO Implement undo_clicked
+    def undo_clicked(self):
+        pass
+
+    def main(self):
+        pygame.init()
+        pygame.font.init()
+        self.load_images()
+        timer = pygame.time.Clock()
+        pygame.display.set_caption('Chess App')
+
+        while self.running:
+            timer.tick(self.fps)
+            self.screen.fill(self.BEIGE)
+            self.draw_board()
```

Comment:
Again as somewhere above - there should be some general GameController, that would control the drawings etc.